classes/class-wc-connect-shipping-method.php

```diff
@@ -349,35 +349,66 @@ public function calculate_shipping( $package = array() ) {
 				$packaging_lookup = $this->service_settings_store->get_package_lookup();
 
 				foreach ( (array) $instance->rates as $rate_idx => $rate ) {
-					$package_names = array();
-					$service_ids   = array();
+					$package_summaries = array();
+					$service_ids       = array();
+
+					$dimension_unit      = get_option( 'woocommerce_dimension_unit' );
+					$weight_unit         = get_option( 'woocommerce_weight_unit' );
+					$measurements_format = '%s x %s x %s ' . $dimension_unit . ', %s ' . $weight_unit;
 
 					foreach ( $rate->packages as $rate_package ) {
-						$package_format = '';
-						$items          = array();
 						$service_ids[]  = $rate_package->service_id;
 
+						$item_product_ids = array();
 						foreach ( $rate_package->items as $package_item ) {
+							$item_product_ids[] = $package_item->product_id;
+						}
+
+						$product_summaries = array();
+						$product_counts = array_count_values( $item_product_ids );
+						foreach ( $product_counts as $product_id => $count ) {
 							/** @var WC_Product $product */
-							$product = $this->lookup_product( $package, $package_item->product_id );
+							$product = $this->lookup_product( $package, $product_id );
 							if ( $product ) {
-								$items[] = WC_Connect_Compatibility::instance()->get_product_name( $product );
+								$item_name = WC_Connect_Compatibility::instance()->get_product_name( $product );
+								$item_measurements = sprintf(
+									$measurements_format,
+									$package_item->length,
```

Comment:
This causes a bug where the wrong measurements show up for the product. `$package_item` is left over from the previous loop and shouldn't be used here.

Reply:
Fixed in https://github.com/Automattic/woocommerce-services/pull/1318/commits/b879a615aa85f34dd4bfe207d5ec784b7fe9f207.

Assuming items with the same product ID would always have the same measurements.

May also have been fixed by getting measurements directly of the product, but this would be less useful than the values from the server.